frontend/lib/qr_code_scanner/qr_code_scanner_page.dart

```diff
@@ -13,7 +13,7 @@ class QrCodeScannerPage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return QrCodeScanner(
-      onCodeScanned: (code) async => await _onCodeScanned(context, code),
+      onCodeScanned: (code) async => _onCodeScanned(context, code),
```

Comment:
Crazy, this seems wrong. Why don't we have to await `_onCodeScanned`? Is it necessary that `onCodeScanned` is async?

Reply:
This is probably just personal preference, I think we had the same discussion in integreat a while back with js. The rule used is this one: https://dart-lang.github.io/linter/lints/unnecessary_await_in_return.html
Basically it is just not necessary to await an async functio if it is returned since the return value is a Future already.
So if you want we can revert this.

Reply:
Both is fine for me, but I generally prefer lints which yield shorter and less-boilerplate code. So I'm fine with disallowing await here.

Reply:
Do we have a lint for "unhandled promises". So for example what happens if you remove the "async" here? Does it show an error like when you add the "await"?

Reply:
Yes, there is an error if async is removed. I'd also favor the current with no `await` if returned. So will keep as it is.